src/compiler/checker.ts

```diff
@@ -1605,6 +1605,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         getBaseTypes,
         getBaseTypeOfLiteralType,
         getWidenedType,
+        getWidenedLiteralType,

```

Comment:
I tried to see if this could be removed and just use `getWidenedType`, but that didn't quite work, which is a shame.

Reply:
Yeah I tried that as well, and I noticed that the dts emit/ Typechecker was explicitly calling getWidenedLiteralType as well so I made the fixer call it instead.